fennel/client/client.py

```diff
@@ -383,8 +383,17 @@ def delete_branch(self, name: str):
         name (str): The name of the branch to delete.
 
         """
+        if name == _MAIN_BRANCH:
+            raise Exception("Cannot delete the main branch.")
+
+        cur_branch = self._branch
+        self.checkout(name)
```

Comment:
we don't need this, right?

Reply:
We checkout the 'name' branch delete it and then checkout the current branch